libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -379,27 +379,26 @@ string YulUtilFunctions::roundUpFunction()
 	});
 }
 
-string YulUtilFunctions::overflowCheckedUIntAddFunction(size_t _bits)
+string YulUtilFunctions::overflowCheckedIntAddFunction(IntegerType const& _type)
 {
-	solAssert(0 < _bits && _bits <= 256 && _bits % 8 == 0, "");
-	string functionName = "checked_add_uint_" + to_string(_bits);
+	string functionName = "checked_add_" + _type.identifier();
 	return m_functionCollector->createFunction(functionName, [&]() {
 		return
 			Whiskers(R"(
 			function <functionName>(x, y) -> sum {
-				<?shortType>
-					let mask := <mask>
-					sum := add(and(x, mask), and(y, mask))
-					if and(sum, not(mask)) { revert(0, 0) }
-				<!shortType>
-					sum := add(x, y)
-					if lt(sum, x) { revert(0, 0) }
-				</shortType>
+				<?signed>
+				if and(iszero(slt(x, 0)), sgt(y, sub(<maxValue>, x))) { revert(0, 0) }
```

Comment:
I might help if you add a comment above here like 
```
if x >= 0 && y > (maxVal - x) -> OVERFLOW
if x < 0 && y < (minVal - x)  -> OVERFLOW
```

Reply:
Might even help to distinguish between "overflow" and "underflow".